.github/workflows/nightly-upstream-snapshot-build.yml

```
@@ -94,19 +102,319 @@ jobs:
+        with:
+          fetch-depth: 0
+          path: aws-otel-java-instrumentation
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v4
+        with:
+          go-version: '~1.18.9'
+
+      - name: Create test batch key values
+        id: set-batches
+        run: |
+          cd testing-framework/tools/batchTestGenerator
+          go build
+          ./batchTestGenerator github --testCaseFilePath=$GITHUB_WORKSPACE/aws-otel-java-instrumentation/.github/workflows/testcases.json --maxBatch=${{ env.NUM_BATCHES }} \
+            --include=EKS_ADOT_OPERATOR,EKS_ADOT_OPERATOR_ARM64
+
+      - name: List testing suites
+        run: |
+          echo ${{ steps.set-batches.outputs.batch-keys }}
+          echo ${{ steps.set-batches.outputs.batch-values }}
+
+  # job to run the e2e integration tests
+  run-batch-job:
+    runs-on: ubuntu-latest
+    needs: [ build, create-test-ref, get-testing-suites ]
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.get-testing-suites.outputs.test-case-batch-key) }}
+
+    steps:
+      # required for versioning
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          path: aws-otel-java-instrumentation
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+
+      - name: Set up terraform
+        uses: hashicorp/setup-terraform@v3
+        with:
+          terraform_version: "~1.5"
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.JAVA_INSTRUMENTATION_INTEG_TEST_ARN}}
+          aws-region: us-west-2
+          # 4 hours
+          role-duration-seconds: 14400
+
+      - name: Checkout testing framework
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.TESTING_FRAMEWORK_REPO }}
+          path: testing-framework
+          ref: ${{ needs.create-test-ref.outputs.testRef }}
+
+      - name: create test-case-batch file
+        run: |
+          jsonStr='${{ needs.get-testing-suites.outputs.test-case-batch-value }}'
+          jsonStr="$(jq -r '.${{ matrix.BatchKey }} | join("\n")' <<< "${jsonStr}")"
+          echo "$jsonStr" >> testing-framework/terraform/test-case-batch
+          cat testing-framework/terraform/test-case-batch
+
+      - name: Get TTL_DATE for cache
+        id: date
+        run: echo "ttldate=$(date -u -d "+7 days" +%s)" >> $GITHUB_OUTPUT
+
+      - name: run tests
+        run: |
+          export TTL_DATE=${{ steps.date.outputs.ttldate }} 
+          export TF_VAR_java_auto_instrumentation_tag=${{ needs.build.outputs.time_stamp_tag }}
+          export TF_VAR_java_auto_instrumentation_repository="${{ env.IMAGE_NAME }}"
+          export DDB_BATCH_CACHE_SK=${{ needs.build.outputs.time_stamp_tag }}
+          cd testing-framework/terraform
+          make execute-batch-test
```

Instead of using `cd`, use `working-directory`.
```suggestion
      - name: run tests
        working-directory: testing-framework/terraform
        run: |
          export TTL_DATE=${{ steps.date.outputs.ttldate }} 
          export TF_VAR_java_auto_instrumentation_tag=${{ needs.build.outputs.time_stamp_tag }}
          export TF_VAR_java_auto_instrumentation_repository="${{ env.IMAGE_NAME }}"
          export DDB_BATCH_CACHE_SK=${{ needs.build.outputs.time_stamp_tag }}
          make execute-batch-test
```

---

Thanks. Updated.